docs/BMI-Installation.md

```diff
@@ -0,0 +1,231 @@
+iptables -I INPUT -p tcp -m tcp --dport 3260 -j ACCEPT
+service iptables save
+
+systemctl start tgtd
+chkconfig tgtd on
+systemctl status tgtd
+```
+
+For more information visit their [website](http://stgt.sourceforge.net/) and the [Quick Start Guide](https://fedoraproject.org/wiki/Scsi-target-utils_Quickstart_Guide)
+
+#### DHCP server
+We support `dnsmasq` for this. 
+
+`$ yum install dnsmasq`
+
+For more information about dnsmasq, you can look at the [wiki](https://wiki.debian.org/HowTo/dnsmasq)
+
+#### Hardware Isolation Layer (HIL)
+
+To setup HIL, you can read the [HIL documentation](http://hil.readthedocs.io/en/latest/)
+
+### Installling
+* Clone this repository to your home folder and change to `dev` branch
+```
+$ git clone https://github.com/CCI-MOC/ims
+$ git checkout dev
+```
+* Run setup.py and install python-cephlibs
+```
+$ sudo python setup.py install
+$ pip install python-cephlibs
+```
+
+That's it. Installation is done!
+***
+
+## Configuration
+
+The template for the config is [here](https://github.com/CCI-MOC/ims/blob/dev/bmi_config.cfg)
+
+This section describes each config section along with sample  
+
+**bmi section**  
+* uid is used to prevent clashing of BMIs that use the same Ceph Pool. It is the responsibility of the admin to ensure that uid is unique.
+* service should be set to True if BMI is deployed as a service. Set it to False for now as there are some glitches.
+```
+# this section is for basic bmi settings
+[bmi]
+# uid is given so that images dont clash in ceph pool
+uid = 5
+service = False
+```  
+
+**db section**  
+* url is the path of the sqlite db
+```
+# this section is for db settings
+[db]
+url = /home/user/bmi.db
+```  
+
+**filesystem section**  
+Tells which Filesystem to use (right now, we only support ceph)
+
+* id is the ceph id which has access to the pool
+* pool is the ceph pool to use for storing images
+* conf_file is the path to the Ceph conf file
+* keyring is the path to the keyring file
+```
+# This section is filesystem related config
+[fs]
+id = bmi
+pool = bmi
+conf_file = /etc/ceph/ceph.conf
+keyring = /etc/ceph/bmi.key
+```
+
+**Driver section**
+* net_isolator is the name of the network isolator to use. it connects the
+* node to a network on which BMI can provision. We use HIL for network
+* isolation. [Link to HIL](http://hil.readthedocs.io/en/latest/)
+* iscsi is the iscsi driver to load
+* fs is the filesystem to use
+```
+# This section is for hil related config
+[driver]
+fs = ims.einstein.fs.ceph.driver
+net_isolator = not used
+iscsi = ims.einstein.tgt.driver
+```  
+
+**iscsi section**
+* ip is the ip of iscsi server on the provisioning network
+* password is the sudo password for the VM (Will be removed)
+```
+# This section is for iscsi related config
+[iscsi]
+ip = 127.0.0.1
+password = password
+```  
+
+**rpc section**
+* name_server is the ip and port Name Server binds to
+* rpc_server is the ip and port RPC Server binds to (The end which calls einstein)
+```
```

Comment:
We should talk about pyro or atleast have a link to it.(https://pypi.python.org/pypi/Pyro4)